Automattic/jetpack
# Boost: rearrange how the File_Storage invalidate() function operates

Instead of having 3 invalidate functions, run everything through one, using a filter to select what to do with the files and directories.

## Proposed changes:
* Remove the other invalidate functions from File_Storage.
* Add a filter to invalidate()
* Update delete_directory so it uses the filter to operate differently. Avoid recursion and use PHP's RecursiveDirectoryIterator class.

### Other information:


## Jetpack product discussion
pc9hqz-2vF-p2

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
Apply PR.
Edit a post and watch error log to make sure files were deleted.
Post a comment in a private window, and do not click "remember me" checkbox. If moderated, confirm that only your own cache file was deleted. Might be tricky to test as it checks if COOKIE array is empty, and then refuses to delete the cache as it's the anon user.
